test/RunSpec.hs

```diff
@@ -102,6 +102,19 @@ spec = do
         , "  location: https://github.com/hspec/hspec"
         ]
 
+    it "includes source repository with subdir" $ do
+      renderPackage 0 [] package {packageSourceRepository = Just ("https://github.com/hspec/hspec", Just "driver")} `shouldBe` unlines [
+          "name: foo"
+        , "version: 0.0.0"
+        , "build-type: Simple"
+        , "cabal-version: >= 1.10"
+        , ""
+        , "source-repository head"
+        , "  type: git"
+        , "  location: https://github.com/hspec/hspec"
+        , "  subdir: driver"
+        ]
```

Comment:
Do you think we could test `renderSourceRepository` independently?  I think the current testing approach does not scale.